Add airflow GitHub script

jhtimmins opened the pull request:
This adds a Github counterpart to the airflow-jira script that generates comparisons between releases as well as changelogs.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

jhtimmins committed:
Add script so changelog can be created against GitHub.

jhtimmins committed:
Make changelog script point to Github instead of Jira.

kaxil committed:
Fix static check

kaxil committed:
Allow passing Github Token Via Environment Variable

kaxil commented on dev/airflow-github:
```diff
@@ -0,0 +1,218 @@
+            if event.event == 'merged':
+                commit = event.commit_id
+        if not commit:
+            continue
```
Because of this the PRs/Issues that are not merged yet won't show in the list

kaxil commented on dev/airflow-github:
```diff
@@ -0,0 +1,218 @@
+        for event in issue.get_events():
+            # Checks if event is a commit getting merged into master.
+            if event.event == 'merged':
+                commit = event.commit_id
```
This loop is time-consuming, takes a good amount of time on my machine to load all commits.

kaxil replied:
I will push some improvements

kaxil replied:
Fixed in https://github.com/apache/airflow/pull/9270/commits/f93f0ee9832614921436e669141fc779ad5a18e6

kaxil committed:
Fix printing all issues

kaxil committed:
Improve Performance of the script

- Fetch the commit in Master locally first without using Github API.
- If not present, find the `merge_commit_sha` using the Github API

jhtimmins commented on dev/airflow-github:
```diff
@@ -93,6 +93,32 @@ def get_issue_type(issue):
+        if commit:
+            return commit
+        else:
+            pr: PullRequest = issue.as_pull_request()
```
this is great. really like this optimization.

jhtimmins committed:
Remove unnecessary method get_commits_to_current_branch.
